File: linked_lists/find_node.java
class ListNode
{
    int val;
    ListNode next;
    ListNode (int val)
    {
        this.val = val;
    }
}

ListNode findNode(ListNode head, int k)
{
    ListNode slow = head;
    ListNode fast = head;

    for (int i = 0; i < k; i++)
    {
        fast = fast.next; 
    }

    while (fast != null)
    {
        fast = fast.next;
        slow = slow.next;
    }

    return slow;
}

File: linked_lists/find_node.js
class ListNode
{
    constructor(val)
    {
        this.val = val;
        this.next = null;
    }
}

const findNode = (head, k) => {
    let slow = head;
    let fast = head;

    for (let i = 0; i < k; i++)
    {
        fast = fast.next;
    }

    while (fast)
    {
        fast = fast.next;
        slow = slow.next;
    }

    return slow;
}